tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -311,16 +312,19 @@ private void UpdateStatus(bool fromRemoteConfig = false)
             {
                 if (_settings.Enabled)
                 {
-                    _waf?.Dispose();
-
-                    _waf = Waf.Waf.Create(_settings.ObfuscationParameterKeyRegex, _settings.ObfuscationParameterValueRegex, _settings.Rules, _remoteRulesJson);
-                    if (_waf?.InitializedSuccessfully ?? false)
+                    _wafInitializationResult = Waf.Waf.Create(_settings.ObfuscationParameterKeyRegex, _settings.ObfuscationParameterValueRegex, _settings.Rules, _remoteRulesJson);
+                    if (_wafInitializationResult.Success)
                     {
+                        var oldWaf = _waf;
+                        _waf = _wafInitializationResult.Waf;
+                        Log.Information("disposing old waf");
```

Comment:
do we need this log? Maybe make it debug?

Reply:
removed in d358405a702f8b3b907075d68c28bb3146c7a794